### PR TITLE
Search in trash documented

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | https://jira.ez.no/browse/EZP-31650 |
| Versions      | 3.1 |

Documentation for the API part of the feature.

### CODE REVIEW

File: docs/api/public_php_api_managing_content.md [hunk 1]
```diff
@@ -130,7 +130,8 @@ $newParent = $this->locationService->loadLocation($location);
 $this->trashService->recover($trashItem, $newParent);
 ```
 
-To sort Trash items by date, use [`TrashService::findTrashItems`.](https://github.com/ezsystems/ezplatform-kernel/blob/v1.0.2/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/DateTrashed.php)
+You can also search through Trash items and sort the results based on several public PHP API search criteria and sort clauses that have been exposed for `TrashService` queries.
```
Comment (adamwojs) [inline docs/api/public_php_api_managing_content.md:133]:
```suggestion
You can also search through Trash items and sort the results using several public PHP API search criteria and sort clauses that have been exposed for `TrashService` queries.
```

File: docs/guide/search/search_in_trash_reference.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+|[ContentTypeId](criteria_reference/contenttypeid_criterion.md)|ID of the Content Item's Content Type|Content and Location|
+|[DateMetadata](criteria_reference/datemetadata_criterion.md)|The date when Content Item was created or last modified|Content and Location|
+|[SectionId](criteria_reference/sectionid_criterion.md)|ID of the Section the Content Item was assigned to|Content and Location|
+|[UserMetadata](criteria_reference/usermetadata_criterion.md)|The creator or modifier of a Content Item|Content and Location|
+
+## Supported logical operators
+
+|Search Criterion|Search based on|Search type|
+|-----|-----|-----|
+|[LogicalAnd](criteria_reference/logicaland_criterion.md)|Implements a logical AND Criterion. It matches if ALL of the provided Criteria match.|Content and Location|
```
Comment (adamwojs) [inline docs/guide/search/search_in_trash_reference.md:19]: Actually, others logical operators should be supported as well.

Comment (adamwojs) [reply docs/guide/search/search_in_trash_reference.md:19]: FYI: ezsystems/ezplatform-kernel#87

File: docs/api/public_php_api_search.md [hunk 3]
```diff
@@ -253,6 +253,17 @@ $query->sortClauses = [
 
     For the full list and details of available Sort Clauses, see [Sort Clause reference](../guide/search/sort_clause_reference.md).
 
+## Searching in trash
+
+You can also search the trash.
+To submit a query for Content Items that are held in trash, use the `TrashService::findInTrash` method.
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_search.md:259]:
```suggestion
To search the trash use the `TrashService::findInTrash` method, submit a query for Content items that are held in trash.
```

File: docs/guide/search/search_in_trash_reference.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Searching in trash reference
+
+When you [search for Content Items that are held in trash](../../api/public_php_api_search.md#searching-in-trash), you can apply only a limited set of Search Criteria and Sort Clauses.
+They can be used by `\eZ\Publish\API\Repository\TrashService::findTrashItems` only.
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:4]:
```suggestion
They can be used by `eZ\Publish\API\Repository\TrashService::findTrashItems` only.
```

File: docs/guide/search/search_in_trash_reference.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+|[LogicalAnd](criteria_reference/logicaland_criterion.md)|Implements a logical AND Criterion. It matches if ALL of the provided Criteria match.|Content and Location|
+|[LogicalNot](criteria_reference/logicalor_criterion.md)|Implements a logical NOT Criterion. It matches if the provided Criterion doesn't match.|Content and Location|
+|[LogicalOr](criteria_reference/logicalor_criterion.md)|Implements a logical OR Criterion. It matches if at least one of the provided Criteria matches.|Content and Location|
+## Supported Sort Clauses
+
+| Sort Clause | Sorting based on | Search type|
+|-----|-----|-----|
+|[ContentName](sort_clause_reference/contentname_sort_clause.md)|Content names|Content and Location|
+|[ContentTypeName](sort_clause_reference/contenttypename_sort_clause.md)|Name of the Content Item's Content Type|Trash only|
+|[DateTrashed](sort_clause_reference/datetrashed_sort_clause.md)|Date when content was sent to trash|Trash only|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:30]:
```suggestion
|[DateTrashed](sort_clause_reference/datetrashed_sort_clause.md)|Date when the Content item was sent to trash|Trash only|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+|[Depth](sort_clause_reference/depth_sort_clause.md)|Location depth in the Content tree|Location only|
+|[Path](sort_clause_reference/path_sort_clause.md)|PathString of the Location|Location only|
+|[Priority](sort_clause_reference/priority_sort_clause.md)|Location priority|Location only|
+|[SectionName](sort_clause_reference/sectionname_sort_clause.md)|Name of the Section that content was assigned to|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:34]:
```suggestion
|[SectionName](sort_clause_reference/sectionname_sort_clause.md)|Name of the Section the Content item was assigned to|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+|Search Criterion|Search based on|Search type|
+|-----|-----|-----|
+|[LogicalAnd](criteria_reference/logicaland_criterion.md)|Implements a logical AND Criterion. It matches if ALL of the provided Criteria match.|Content and Location|
+|[LogicalNot](criteria_reference/logicalor_criterion.md)|Implements a logical NOT Criterion. It matches if the provided Criterion doesn't match.|Content and Location|
+|[LogicalOr](criteria_reference/logicalor_criterion.md)|Implements a logical OR Criterion. It matches if at least one of the provided Criteria matches.|Content and Location|
+## Supported Sort Clauses
+
+| Sort Clause | Sorting based on | Search type|
+|-----|-----|-----|
+|[ContentName](sort_clause_reference/contentname_sort_clause.md)|Content names|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:28]:
```suggestion
|[ContentName](sort_clause_reference/contentname_sort_clause.md)|Content item's name|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+|[ContentTypeName](sort_clause_reference/contenttypename_sort_clause.md)|Name of the Content Item's Content Type|Trash only|
+|[DateTrashed](sort_clause_reference/datetrashed_sort_clause.md)|Date when content was sent to trash|Trash only|
+|[Depth](sort_clause_reference/depth_sort_clause.md)|Location depth in the Content tree|Location only|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:31]:
```suggestion
|[Depth](sort_clause_reference/depth_sort_clause.md)|Location depth in the Content Tree|Location only|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+|[Path](sort_clause_reference/path_sort_clause.md)|PathString of the Location|Location only|
+|[Priority](sort_clause_reference/priority_sort_clause.md)|Location priority|Location only|
+|[SectionName](sort_clause_reference/sectionname_sort_clause.md)|Name of the Section that content was assigned to|Content and Location|
+|[UserLogin](sort_clause_reference/userlogin_sort_clause.md)|Login of the Content Item's creator|Trash only|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:35]:
```suggestion
|[UserLogin](sort_clause_reference/userlogin_sort_clause.md)|Login of the Content item's creator|Trash only|
```

File: docs/guide/search/sort_clause_reference/userlogin_sort_clause.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# UserLogin Sort Clause
+
+The [`UserLogin` Sort Clause](https://github.com/ezsystems/ezplatform-kernel/blob/master/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/UserLogin.php)
+sorts the results of searching in Trash by the login of the Content Item's creator.
```
Comment (MagdalenaZuba) [inline docs/guide/search/sort_clause_reference/userlogin_sort_clause.md:4]:
```suggestion
sorts the results of searching in Trash by the login of the Content item's creator.
```

File: docs/api/public_php_api_search.md [hunk 11]
```diff
@@ -253,6 +253,17 @@ $query->sortClauses = [
...
+Searching in trash supports a limited set of Criteria and Sort Clauses.
+For a list of supported Criteria and Sort Clauses, see [Searching in trash reference](../guide/search/sort_clause_reference.md).
+
+!!! note
+
+    Searching through the trashed Content Items operates directly on the database, therefore you cannot use external search engines, such as Solr or Elasticsearch, and it is impossible to reindex the data.
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_search.md:265]:
```suggestion
    Searching through the trashed Content items operates directly on the database, therefore you cannot use external search engines, such as Solr or Elasticsearch, and it is impossible to reindex the data.
```

File: docs/guide/search/search.md [hunk 12]
```diff
@@ -111,6 +111,17 @@ Available tags for Sort Clause handlers in Legacy Storage Engine are:
...
     You will find all the native handlers and the tags for the Legacy Storage Engine in files located in `eZ/Publish/Core/settings/storage_engines/legacy/`.
 
+!!! tip
+
+    When you search in trash, use the following service tags:
+
+    - for Criterion handlers: `ezplatform.trash.search.legacy.gateway.criterion_handler`
+    - for Sort Clause handlers: `ezplatform.trash.search.legacy.gateway.sort_clause_handler`
+
+    For more information about searching for Content Items in Trash, see [Searching in trash](../../api/public_php_api_search.md#searching-in-trash).
```
Comment (MagdalenaZuba) [inline docs/guide/search/search.md:121]:
```suggestion
    For more information about searching for Content items in Trash, see [Searching in trash](../../api/public_php_api_search.md#searching-in-trash).
```

File: docs/guide/search/search.md [hunk 13]
```diff
@@ -111,6 +111,17 @@ Available tags for Sort Clause handlers in Legacy Storage Engine are:
...
+
+    For more information about the Criteria and Sort Clauses that are supported when searching for trashed Content Items, see [Searching in trash reference](search_in_trash_reference.md).
```
Comment (MagdalenaZuba) [inline docs/guide/search/search.md:123]:
```suggestion
    For more information about the Criteria and Sort Clauses that are supported when searching for trashed Content items, see [Searching in trash reference](search_in_trash_reference.md).
```

File: docs/guide/search/search_in_trash_reference.md [hunk 14]
```diff
@@ -0,0 +1,35 @@
+# Searching in trash reference
+
+When you [search for Content Items that are held in trash](../../api/public_php_api_search.md#searching-in-trash), you can apply only a limited set of Search Criteria and Sort Clauses.
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:3]:
```suggestion
When you [search for Content items that are held in trash](../../api/public_php_api_search.md#searching-in-trash), you can apply only a limited set of Search Criteria and Sort Clauses.
```

File: docs/guide/search/search_in_trash_reference.md [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+They can be used by `\eZ\Publish\API\Repository\TrashService::findTrashItems` only.
+
+## Supported Search Criteria
+
+| Search Criterion | Search based on | Search type |
+|-----|-----|-----|
+|[ContentTypeId](criteria_reference/contenttypeid_criterion.md)|ID of the Content Item's Content Type|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:10]:
```suggestion
|[ContentTypeId](criteria_reference/contenttypeid_criterion.md)|ID of the Content item's Content Type|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+|[DateMetadata](criteria_reference/datemetadata_criterion.md)|The date when Content Item was created or last modified|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:11]:
```suggestion
|[DateMetadata](criteria_reference/datemetadata_criterion.md)|The date when Content item was created or last modified|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+|[MatchAll](criteria_reference/matchall_criterion.md)|Returns all search results|Content and Location|
+|[MatchNone](criteria_reference/matchnone_criterion.md)|Returns no search results|Content and Location|
+|[SectionId](criteria_reference/sectionid_criterion.md)|ID of the Section the Content Item was assigned to|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:14]:
```suggestion
|[SectionId](criteria_reference/sectionid_criterion.md)|ID of the Section the Content item was assigned to|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+|[UserMetadata](criteria_reference/usermetadata_criterion.md)|The creator or modifier of a Content Item|Content and Location|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:15]:
```suggestion
|[UserMetadata](criteria_reference/usermetadata_criterion.md)|The creator or modifier of a Content item|Content and Location|
```

File: docs/guide/search/search_in_trash_reference.md [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+|-----|-----|-----|
+|[LogicalAnd](criteria_reference/logicaland_criterion.md)|Implements a logical AND Criterion. It matches if ALL of the provided Criteria match.|Content and Location|
+|[LogicalNot](criteria_reference/logicalor_criterion.md)|Implements a logical NOT Criterion. It matches if the provided Criterion doesn't match.|Content and Location|
+|[LogicalOr](criteria_reference/logicalor_criterion.md)|Implements a logical OR Criterion. It matches if at least one of the provided Criteria matches.|Content and Location|
+## Supported Sort Clauses
+
+| Sort Clause | Sorting based on | Search type|
+|-----|-----|-----|
+|[ContentName](sort_clause_reference/contentname_sort_clause.md)|Content names|Content and Location|
+|[ContentTypeName](sort_clause_reference/contenttypename_sort_clause.md)|Name of the Content Item's Content Type|Trash only|
```
Comment (MagdalenaZuba) [inline docs/guide/search/search_in_trash_reference.md:29]:
```suggestion
|[ContentTypeName](sort_clause_reference/contenttypename_sort_clause.md)|Name of the Content item's Content Type|Trash only|
```

File: docs/guide/search/sort_clause_reference/datetrashed_sort_clause.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
+# DateTrashed Sort Clause
+
+The [`DateTrashed` Sort Clause](https://github.com/ezsystems/ezplatform-kernel/blob/v1.0.2/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/DateTrashed.php)
```
Comment (MagdalenaZuba) [inline docs/guide/search/sort_clause_reference/datetrashed_sort_clause.md:3]:
```suggestion
The [`DateTrashed` Sort Clause](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0-rc2/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/DateTrashed.php)
```

File: docs/guide/search/sort_clause_reference/userlogin_sort_clause.md [hunk 21]
```diff
@@ -0,0 +1,15 @@
+# UserLogin Sort Clause
+
+The [`UserLogin` Sort Clause](https://github.com/ezsystems/ezplatform-kernel/blob/master/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/UserLogin.php)
```
Comment (MagdalenaZuba) [inline docs/guide/search/sort_clause_reference/userlogin_sort_clause.md:3]:
```suggestion
The [`UserLogin` Sort Clause](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0-rc2/eZ/Publish/API/Repository/Values/Content/Query/SortClause/Trash/UserLogin.php)
```

**Pull request merged**